src/utils.js

```diff
@@ -0,0 +1,7 @@
+export function enqueueJob(callback) {
+  if (typeof queueMicrotask !== 'undefined' && /native code/.test(queueMicrotask.toString())) {
```

Comment:
Curious, why this implementation, and not something simpler, like this?

```js
typeof queueMicrotask === 'function'
```

Reply:
This would also work, but this way you assume that everyone abides by the rules. I could do something like this and this check of yours would still pass:

`window.queueMicrotask = () => 'naughty'`

Testing whether it outputs **native code** in the toString method makes it 99.99% sure the thing is what we think it actually is.

If I did `window.queueMicrotask = Function.prototype`, my check would also fail, but there is no way around this because: JavaScript.

Reply:
Understood. I guess then I don't see the risk here that requires this level of validation. If someone adds `window.queueMicrotask = () => 'naughty'`, then...well...their progress indicator will be broken.

The queue-microtask NPM package (which is downloaded almost 1/2 million times a month) uses the simpler approach as well: https://github.com/feross/queue-microtask/blob/662b71d9a666e964e6d97be22a7468c014bd740c/index.js#L4

If that's sufficient for that package, it would probably be sufficient for us, no?

Totally not trying to be difficult here. I just have to maintain this code moving forward...and simpler === better for me. ❤️

Reply:
As you wish.